backend/src/main/java/com/redhat/cloud/notifications/db/repositories/TemplateRepository.java

```diff
@@ -0,0 +1,338 @@
+            aggregationEmailTemplate.filterOutTemplates();
+        }
+        return aggregationEmailTemplates;
+    }
+
+    public List<AggregationEmailTemplate> findAggregationEmailTemplatesByApplication(UUID appId) {
+        String hql = "FROM AggregationEmailTemplate t JOIN FETCH t.application a WHERE a.id = :appId";
+        List<AggregationEmailTemplate> aggregationEmailTemplates = entityManager.createQuery(hql, AggregationEmailTemplate.class)
+                .setParameter("appId", appId)
+                .getResultList();
+        for (AggregationEmailTemplate aggregationEmailTemplate : aggregationEmailTemplates) {
+            // The full application isn't needed in the REST response.
+            aggregationEmailTemplate.filterOutApplication();
+            // The full templates aren't needed in the REST response.
+            aggregationEmailTemplate.filterOutTemplates();
+        }
+        return aggregationEmailTemplates;
+    }
+
+    public AggregationEmailTemplate findAggregationEmailTemplateById(UUID id) {
+        String hql = "FROM AggregationEmailTemplate t JOIN FETCH t.subjectTemplate JOIN FETCH t.bodyTemplate " +
+                "LEFT JOIN FETCH t.application WHERE t.id = :id";
+        try {
+            AggregationEmailTemplate template = entityManager.createQuery(hql, AggregationEmailTemplate.class)
+                    .setParameter("id", id)
+                    .getSingleResult();
+            // The full application isn't needed in the REST response.
+            template.filterOutApplication();
+            return template;
+        } catch (NoResultException e) {
+            throw new NotFoundException("Aggregation email template not found");
+        }
+    }
+
+    @Transactional
+    public boolean updateAggregationEmailTemplate(UUID id, AggregationEmailTemplate template) {
+        String hql = "UPDATE AggregationEmailTemplate SET application = :app, subjectTemplate = :subjectTemplate, " +
+                "subscriptionType = :subscriptionType, bodyTemplate = :bodyTemplate WHERE id = :id";
+
+        Application app;
+        if (template.getApplicationId() == null) {
+            app = null;
+        } else {
+            app = findApplication(template.getApplicationId());
+        }
+        Template subjectTemplate = findTemplate(template.getSubjectTemplateId(), SUBJECT_NOT_FOUND);
+        Template bodyTemplate = findTemplate(template.getBodyTemplateId(), BODY_NOT_FOUND);
+
+        int rowCount = entityManager.createQuery(hql)
+                .setParameter("app", app)
+                .setParameter("subscriptionType", template.getSubscriptionType())
+                .setParameter("subjectTemplate", subjectTemplate)
+                .setParameter("bodyTemplate", bodyTemplate)
+                .setParameter("id", id)
+                .executeUpdate();
+        return rowCount > 0;
+    }
+
+    @Transactional
+    public boolean enableOrDisableAggregationEmailTemplate(UUID id, boolean enabled) {
```

Comment:
The name seems odd - but not really important

Reply:
It is long and not pretty, but it describes exactly what the method does 😃 I'm OK to change it if you have a better idea.

Reply:
This method no longer exists.

Reply:
That was a clever way to avoid naming something :)
![image](https://user-images.githubusercontent.com/3845764/161833369-0ec235e9-5bdf-486c-b3f2-802e42599797.png)